src/SHA1.elm

```diff
@@ -343,27 +338,33 @@ reduceChunkHelp (State initial) b1 b2 b3 b4 b5 b6 b7 b8 b9 b10 b11 b12 b13 b14 b
     let
         initialDeltaState =
             DeltaState initial
-                |> calculateDigestDeltas 0 b1
-                |> calculateDigestDeltas 1 b2
-                |> calculateDigestDeltas 2 b3
-                |> calculateDigestDeltas 3 b4
-                |> calculateDigestDeltas 4 b5
-                |> calculateDigestDeltas 5 b6
-                |> calculateDigestDeltas 6 b7
-                |> calculateDigestDeltas 7 b8
-                |> calculateDigestDeltas 8 b9
-                |> calculateDigestDeltas 9 b10
-                |> calculateDigestDeltas 10 b11
-                |> calculateDigestDeltas 11 b12
-                |> calculateDigestDeltas 12 b13
-                |> calculateDigestDeltas 13 b14
-                |> calculateDigestDeltas 14 b15
-                |> calculateDigestDeltas 15 b16
+                |> calculateDigestDeltasChunk 0 b1
+                |> calculateDigestDeltasChunk 1 b2
+                |> calculateDigestDeltasChunk 2 b3
+                |> calculateDigestDeltasChunk 3 b4
+                |> calculateDigestDeltasChunk 4 b5
+                |> calculateDigestDeltasChunk 5 b6
+                |> calculateDigestDeltasChunk 6 b7
+                |> calculateDigestDeltasChunk 7 b8
+                |> calculateDigestDeltasChunk 8 b9
+                |> calculateDigestDeltasChunk 9 b10
+                |> calculateDigestDeltasChunk 10 b11
+                |> calculateDigestDeltasChunk 11 b12
+                |> calculateDigestDeltasChunk 12 b13
+                |> calculateDigestDeltasChunk 13 b14
+                |> calculateDigestDeltasChunk 14 b15
+                |> calculateDigestDeltasChunk 15 b16
 
         (DeltaState { a, b, c, d, e }) =
             reduceWords 0 initialDeltaState b1 b2 b3 b4 b5 b6 b7 b8 b9 b10 b11 b12 b13 b14 b15 b16
     in
-    State (Tuple5 (initial.a + a) (initial.b + b) (initial.c + c) (initial.d + d) (initial.e + e))
+    State
+        { a = initial.a + a
+        , b = initial.b + b
+        , c = initial.c + c
+        , d = initial.d + d
+        , e = initial.e + e
+        }
```

Comment:
A plain record is much faster than using the alias (which is a function internally, the record literal is really an object literal in JS)